airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -403,16 +407,17 @@ def consume_logs(
             """
             last_captured_timestamp = None
             try:
-                logs = self.read_pod_logs(
-                    pod=pod,
-                    container_name=container_name,
-                    timestamps=True,
-                    since_seconds=(
-                        math.ceil((pendulum.now() - since_time).total_seconds()) if since_time else None
-                    ),
-                    follow=follow,
-                    post_termination_timeout=termination_timeout,
-                )
+                if not logs:
```

Comment:
I'm not sure if reusing the same `PodLogsConsumer` would fix the issue.

The next time when you reuse it, it will call the method `__iter__` and then all the other methods as if it's a new `PodLogsConsumer` instance.

Did you have the opportunity to test it?

Reply:
Okay, I will request again after writing a unit test.

Reply:
> Did you have the opportunity to test it?

Yes, I am using the airflow applied this patch and checked that there is no duplicated logs.

---

I added a test case to check whether the PodLogsConsumer is reused correctly and there are no duplicated logs both.

Reply:
@hussein-awala I am waiting again :)

Reply:
I am still waiting! Would you review again? @hussein-awala